gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -126,7 +257,395 @@ static void
+          return FALSE;
+        }
+
+        if (0 != nnstreamer_query_receive (self->src_conn, &cmd_buf,
+                DEFAULT_TIMEOUT_MS)) {
+          nns_loge ("Failed to receive response from the query server.");
+          return FALSE;
+        }
+
+        if (cmd_buf.cmd == _TENSOR_QUERY_CMD_RESPOND_APPROVE) {
+          if (gst_tensors_info_validate (&cmd_buf.data_info.config.info)) {
+            gst_tensors_info_copy (&self->out_config.info,
+                &cmd_buf.data_info.config.info);
+            /** The server's framerate is 0/1, set it the same as the input. */
+            self->out_config.format = cmd_buf.data_info.config.format;
+            self->out_config.rate_n = self->in_config.rate_n;
+            self->out_config.rate_d = self->in_config.rate_d;
+            gst_tensor_query_client_update_caps (self);
+          }
+        } else {
+          /** @todo Retry for info */
+          nns_loge ("Failed to receive approve command.");
+          return FALSE;
+        }
+      }
+
+      self->sink_conn =
+          nnstreamer_query_connect (self->protocol, self->sink_host,
+          self->sink_port, DEFAULT_TIMEOUT_MS);
+      if (!self->src_conn) {
+        nns_loge ("Failed to connect server sink ");
+        return FALSE;
+      }
+
+      return TRUE;
+    }
+    default:
+      break;
+  }
+
+  return gst_pad_event_default (pad, parent, event);
+}
+
+/**
+ * @brief This function handles sink pad query.
+ */
+static gboolean
+gst_tensor_query_client_sink_query (GstPad * pad,
+    GstObject * parent, GstQuery * query)
+{
+  GstTensorQueryClient *self = GST_TENSOR_QUERY_CLIENT (parent);
+
+  GST_DEBUG_OBJECT (self, "Received %s query: %" GST_PTR_FORMAT,
+      GST_QUERY_TYPE_NAME (query), query);
+
+  switch (GST_QUERY_TYPE (query)) {
+    case GST_QUERY_CAPS:
+    {
+      GstCaps *caps;
+      GstCaps *filter;
+
+      gst_query_parse_caps (query, &filter);
+      caps = gst_tensor_query_client_query_caps (self, pad, filter);
+
+      gst_query_set_caps_result (query, caps);
+      gst_caps_unref (caps);
+      return TRUE;
+    }
+    case GST_QUERY_ACCEPT_CAPS:
+    {
+      GstCaps *caps;
+      GstCaps *template_caps;
+      gboolean res = FALSE;
+
+      gst_query_parse_accept_caps (query, &caps);
+      silent_debug_caps (self, caps, "accept-caps");
+
+      if (gst_caps_is_fixed (caps)) {
+        template_caps = gst_pad_get_pad_template_caps (pad);
+
+        res = gst_caps_can_intersect (template_caps, caps);
+        gst_caps_unref (template_caps);
+      }
+
+      gst_query_set_accept_caps_result (query, res);
+      return TRUE;
+    }
+    default:
+      break;
+  }
+
+  return gst_pad_query_default (pad, parent, query);
+}
+
+/**
+ * @brief Get start command buffer
+ */
+static gboolean
+gst_tensor_query_client_handle_cmd_buf (GstTensorQueryClient * self,
+    GstBuffer * buf, TensorQueryCommandData * cmd_buf)
+{
+  GstMemory *in_mem;
+  GstMapInfo in_info;
+  guint i, num_tensors = 0;
```

Comment:
it seems that the value initialization is meaningless.

Reply:
Right. Updated.